Daviddt17/island-android-app
# Ignore events and messages that are invalid

This PR adds basic validation to the clients. I tested posting short blobs, invalid blobs, and blobs of the incorrect type. When the exception occurs, the client disregards the blob.

To test this
- Find the aliases and mailbox being queried by checking the verbose logging in `Rest` and `EventSyncAdapter`.
- Post bad data to the server under those keys (it has to be valid JSON or the server will ignore)
- Refresh the app and make sure it logs the error (level warn) and doesn't blow up
